edge-modules/load-gen/Program.cs

```diff
@@ -34,32 +32,39 @@ static async Task Main()
 
                 (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Logger);
 
-                // setup the message PeriodicTask
-                using (var messageTask = new PeriodicTask(
-                    () => GenerateMessageAsync(moduleClient, batchId),
-                    Settings.Current.MessageFrequency,
-                    Settings.Current.StartDelay,
-                    Logger,
-                    "Generate message"))
+                Logger.LogInformation("Load gen running.");
+                Logger.LogInformation($"Load gen delay start for {Settings.Current.TestStartDelay}.");
+                await Task.Delay(Settings.Current.TestStartDelay);
+
+                DateTime testStartAt = DateTime.UtcNow;
+                long messageIdCounter = 0;
+                while (!cts.IsCancellationRequested && DateTime.UtcNow - testStartAt < Settings.Current.TestDuration)
                 {
-                    Logger.LogInformation("Load gen running.");
-                    await cts.Token.WhenCanceled();
+                    messageIdCounter++;
```

Comment:
Super nit: Start from 1 or 0? :)

Reply:
Also, we should not update the messageIdCounter if the sending fails

Reply:
I want to start id from 1; and counter will increment before calling send method.  It will report to TestAnalyzer (implement soon), for both success and failed calls of SendMessage method.  I expected messageCounter will increase even fail case.

Reply:
let me know if you prefer starting from 0.

Reply:
If we increase the messageCounter in failure case, you will run into the same issue that we have been facing. 
Also, will reporting for each message sent scale? What if we want to use this to stress edgeHub? We will be stressing the Analyzer then as well.

Reply:
In the analyzer there is a trade off here wrt storage. Either we:
1 - Store only a count of status types in-memory which uses a constant amount of memory. We currently use this approach for twins and direct methods.
2 - Store each status in memory so as to make errors more traceable (as we can store individual times). We currently use this approach for messages.

For http request overload, Anca and I have discussed sharding the analyzer in the past which I think could solve the issue if it arises.